wit/core.wit

```diff
@@ -0,0 +1,89 @@
+package wasmcloud:core@0.1.0
+
+interface types {
+  type link-settings =          tuple<string, string>
+  type host-env-values =        tuple<string, string>
+  type actor-links =            list<link-defination>
+  type cluster-issuers =        list<cluster-issuer-key>
+  type cluster-issuer-key =     string 
+  type capability-contract-id = string
+  type blob =                   list<u8>
+  type trace-context =          tuple<string, string>
+
+  enum log-level {
+    trace,
+    debug,
+    info,
+    warn,
+    error,
+  }
+
+  record link-defination {
+    actor-id:     string,
+    provider-id:  string,
+    link-name:    string,
+    contract-id:  string,
+    values:       option<link-settings>,
+  }
+
+  record health-check-response {
+    healthy: bool,
+    message: string,
+  }
+
+  record health-check-request { }
+  
+  record host-data {
+    host-id: string,
+    lattice-rpc-prefix:     string,
+    link-name:              string,
+    lattice-rpc-user-jwt:   string,
+    lattice-rpc-user-seed:  string,
+    lattice-rpc-url:        string,
+    provider-key:           string,
+    invocation-seed:        string,
+    env-values:             host-env-values,
+    instance-id:            string,
+    link-definations:       actor-links,
+    cluster-issuers:        cluster-issuers,
+    config-json:            string,
+    default-rpc-timeout-ms: u64,
+    structured-logging:     bool,
+    log-level:              log-level,
+  }
+
+  record invocation {
+      origin:         wasmcloud-entity,
+      target:         wasmcloud-entity,
+      operation:      string,
+      msg:            blob,
+      id:             string,
+      encoded-claims: string,
+      host-id:        string,
+      content-length: u64,
+      trace-context:  trace-context,
+    }
+
+  record wasmcloud-entity {
```

Comment:
one of the cool things about switching from smithy to wit is that wit supports union types. This struct was created with this shape because we couldn't use discriminated unions.

We could do something like the following for more clarity:

```
 variant wasmcloud-entity {
    actor(string),
    provider(string, string, string)
  }
```
where the provider takes the public key, contract ID, and link name.

Reply:
we could make this even more clear and potentially self-documenting (certainly less error prone) by doing the following:
```
record provider-identifier  {
    public-key: string,
    contract-id: string,
    link-name: string
}
type public-key = string

variant wasmcloud-entity {
  actor(public-key),
  provider(provider-identifier)
}
```